src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -22,147 +23,160 @@ internal class ListPackageCommandRunner : IListPackageCommandRunner
-                ProjectPackagesPrintUtility.PrintSources(listPackageArgs.PackageSources);
-                Console.WriteLine();
+                await GetProjectMetadataAsync(projectPath, listPackageReportModel, msBuild, listPackageArgs);
             }
 
-            foreach (var projectPath in projectsPaths)
-            {
-                //Open project to evaluate properties for the assets
-                //file and the name of the project
-                var project = MSBuildAPIUtility.GetProject(projectPath);
+            // if there is any error then return failure code.
+            int exitCode = (
+                listPackageArgs.Renderer.GetProblems().Any(p => p.ProblemType == ProblemType.Error)
+                || listPackageReportModel.Projects.Where(p => p.ProjectProblems != null).SelectMany(p => p.ProjectProblems).Any(p => p.ProblemType == ProblemType.Error))
+                ? GenericFailureExitCode : GenericSuccessExitCode;
 
-                if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
-                {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_NotPRProject,
-                        projectPath));
-                    Console.WriteLine();
-                    continue;
-                }
+            return (exitCode, listPackageReportModel);
+        }
+
+        private async Task GetProjectMetadataAsync(
+            string projectPath,
+            ListPackageReportModel listPackageReportModel,
+            MSBuildAPIUtility msBuild,
+            ListPackageArgs listPackageArgs)
+        {
+            //Open project to evaluate properties for the assets
+            //file and the name of the project
+            Project project = MSBuildAPIUtility.GetProject(projectPath);
+            var projectName = project.GetPropertyValue(ProjectName);
+            // Project specific data stored in below variable
+            ListPackageProjectModel projectModel = listPackageReportModel.CreateProjectReportData(projectPath: projectPath, projectName);
+
+            if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
+            {
+                projectModel.AddProjectInformation(problemType: ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_NotPRProject, projectPath));
+                return;
+            }
 
-                var projectName = project.GetPropertyValue(ProjectName);
+            var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
 
-                var assetsPath = project.GetPropertyValue(ProjectAssetsFile);
+            // If the file was not found, print an error message and continue to next project
+            if (!File.Exists(assetsPath))
+            {
+                projectModel.AddProjectInformation(ProblemType.Error,
+                    string.Format(CultureInfo.CurrentCulture, Strings.Error_AssetsFileNotFound, projectPath));
```

Comment:
here the comment is definitely not adding anything that is not already obvious from the code.